File: csharp/1464.maximum-product-of-two-elements-in-an-array.cs
/*
 * @lc app=leetcode id=1464 lang=csharp
 *
 * [1464] Maximum Product of Two Elements in an Array
 */

// @lc code=start
public class Solution {
    public int MaxProduct(int[] nums) {
        int f = 0, s = 0;
        foreach (int x in nums) {
            if (x >= f) {
                s = f;
                f = x;
            } else if (x > s) {
                s = x;
            }
        }
        return (f - 1) * (s - 1);
    }
}
// @lc code=end

File: java/1464.maximum-product-of-two-elements-in-an-array.java
/*
 * @lc app=leetcode id=1464 lang=java
 *
 * [1464] Maximum Product of Two Elements in an Array
 */

// @lc code=start
class Solution {
    public int maxProduct(int[] nums) {
        int m = -1, m2 = -1;
        for (int num : nums) {
            if (num > m) {
                m2 = m;
                m = num;
            } else if (num > m2) {
                m2 = num;
            }
        }
        return (m2 - 1) * (m - 1);
    }
}
// @lc code=end

